app.rb

```diff
@@ -1,141 +1,172 @@
-require_relative('./book')
-require_relative('./teacher')
-require_relative('./student')
-require_relative('./rental')
+require 'json'
+require_relative './book'
+require_relative './teacher'
+require_relative './student'
+require_relative './rental'
+require_relative './loaders'
 
 class App
   def initialize
     @persons = []
     @books = []
     @rentals = []
+
+    loader = Loader.new
+
+    loader.load_person_from_file(@persons)
+    loader.load_books_from_file(@books)
+    loader.load_rentals_from_file(@rentals)
   end
 
-  # 1 - list all books
   def books_list
     if @books.length.positive?
       @books.each do |book|
-        puts " Title: #{book.title}, Author: #{book.author}"
+        puts "Title: #{book['title']}, Author: #{book['author']}"
+        puts
       end
     else
       puts 'No books added yet!'
     end
-    run
   end
 
-  # 2 - List all people'
   def person_list
     @persons.each do |individual|
-      # binding.pry
-      puts "[#{individual.class}] Name: #{individual.name}, Age: #{individual.age}"
+      puts "[#{individual['class_name']}],
+        Name: #{individual['name']},
+        ID: #{individual['id']}, Age: #{individual['age']}"
     end
-    run
   end
 
-  # '3 - Create a teacher',
-  def create_teacher
-    print 'Age: '
-    age = gets.chomp
-    print 'Name: '
-    name = gets.chomp
+  def create_teacher(age, name)
     print 'Specialization: '
     specialization = gets.chomp
-    @persons << Teacher.new(specialization, name, age)
+    new_teacher = Teacher.new(specialization, name, age)
+
+    hash = {
+      'class_name' => new_teacher.class.name,
+      'name' => name,
+      'age' => age,
+      'specialization' => new_teacher.specialization,
+      'id' => new_teacher.id
+    }
+    @persons << hash
     puts 'create teacher'
   end
 
-  # '3 - Create a student',
-  def create_student
-    print 'Age: '
-    age = gets.chomp
-    print 'Name: '
-    name = gets.chomp
+  def create_student(age, name)
     print 'Has parent permission? [Y/N]: '
     parent_permission = gets.chomp != 'n'
-    @persons << Student.new(name, age, parent_permission)
+
+    new_student = Student.new(name, age, parent_permission)
+    hash = {
+      'class_name' => new_student.class.name,
+      'name' => name,
+      'age' => age,
+      'parent_permission' => new_student.parent_permission,
```

Comment:
- [ ] I think you have an issue here when I try to make a person to see if it will bve saved to the json files I can't create it so please fix this problem to be able to see the json files 🔥♻💯

![image](https://user-images.githubusercontent.com/58553711/201100263-be0090fa-bd4f-4bd1-b909-4bb02a615b05.png)

Reply:
Noted ✅😊